.github/workflows/release.yml

```diff
@@ -71,16 +79,36 @@ jobs:
           --shard-count "${{matrix.shard_count}}" \
           --shard-index "${{matrix.shard_index}}" \
           --export-tags-file bake-target-tags
-          # push images to image repository
-          if [ -f bake-target-tags ]; then
-            docker image push "$(< bake-target-tags)"
-          fi
-      - name: Set up Cosign
-        uses: sigstore/cosign-installer@v3.1.2
-      - name: Sign images
-        run: |
+          # Push images to image repository
           if [ -f bake-target-tags ]; then
-            cosign sign -y "$(< bake-target-tags)"
+            echo "bake-target-tags: "$(< bake-target-tags)
+            IMAGE_NAME=$(cat bake-target-tags | cut -d ":" -f 1)
+            TAG_NAME=$(cat bake-target-tags | cut -d ":" -f 2)
+            echo "image: $IMAGE_NAME"
+            echo "tag: $TAG_NAME"
+            # Store the output of `docker image push` into a variable, so we can parse it for the digest
+            PUSH_OUTPUT=$(docker image push "$(< bake-target-tags)" 2>&1)
+            echo "$PUSH_OUTPUT"
+            # Obtain the digest of the pushed image from the output of `docker image push`, because signing by tag is deprecated and will be removed from cosign in the future
+            DIGEST=$(echo "$PUSH_OUTPUT" | awk "/: digest: sha256:[a-f0-9]{64} size: [0-9]+$/ { print \$3 }")
+            # Refer to image via its digest (docker.stackable.tech/stackable/airflow@sha256:0a1b2c...)
+            # This generates a signature and publishes it to the registry, next to the image
+            # Uses the keyless signing flow with Github Actions as identity provider
+            cosign sign -y "$IMAGE_NAME@$DIGEST"
+
+            # Push to oci.stackable.tech as well
+            IMAGE_NAME=oci.stackable.tech/stackable/${{ matrix.product }}
```

Comment:
Same as other comment, but just so this isn't forgotten if the other reference changes